### ISSUE TITLE
Python 3.9's builtin collections type hints do not work in pydantic types 

### ISSUE THREAD

**Issue by benzolium** [strawberry-graphql/strawberry#2083]
Python 3.9's builtin collections type hints do not work in pydantic types.
Tested with list.

## Describe the Bug

Here's the test to reproduce.
Just changed line 219 in `tests/experimental/pydantic/test_basic.py`
`List` -> `list`.

```python
def test_list():
    class User(pydantic.BaseModel):
        friend_names: list[str]  # <- List from typings works, but list - does not

    @strawberry.experimental.pydantic.type(User)
    class UserType:
        friend_names: strawberry.auto

    definition: TypeDefinition = UserType._type_definition
    assert definition.name == "UserType"

    [field] = definition.fields

    assert field.python_name == "friend_names"
    assert isinstance(field.type, StrawberryList)
    assert field.type.of_type is str
```

Got:
```
E       TypeError: issubclass() arg 1 must be a class
```

## System Information

 - Operating system: mac oc
 - Strawberry version (if applicable): 0.124.0

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/strawberry-graphql) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/strawberry-graphql/strawberry/issues/2083">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/strawberry-graphql/strawberry/issues/2083/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/strawberry-graphql/strawberry/issues/2083/pledge.svg">
</picture>
</a>

**Comment by benzolium** [comment]
@na86421 that exact error occurs after that patch indeed, good catch!

Though I've tried to checkout on 2495d0994b25e1e7462e59713bc7ba8b02e3a2f0 (commit before that patch) and it still fails but with another error:
```
E           AttributeError: type object 'list' has no attribute 'copy_with'
```

**Comment by DoctorJohn** [comment]
Looks like this has been resolved in #2152 :)
